amethyst_assets/src/loader.rs

```diff
@@ -86,7 +86,7 @@ impl Loader {
     /// Registers an asset and inserts a context into the map.
     pub fn register<A, C>(&mut self, context: C)
         where A: Asset + 'static,
-              C: Context<Asset=A> + 'static,
+              C: Context<Asset=A> + Send + Sync + 'static,
```

Comment:
For an engine with a high emphasis on multi threading I think it's OK to litter Send + Sync restrictions.  This way we can do more with out context types without worrying about breaking code in the future.